pkg/controller/controller.go

```diff
@@ -353,7 +377,20 @@ func (ctrl *resizeController) syncPVC(key string) error {
 			return nil
 		}
 
-		return ctrl.resizePVC(pvc, pv)
+		err := ctrl.resizePVC(pvc, pv)
+		if err != nil {
+			return err
+		}
+	}
+
+	vacName := pvc.Spec.VolumeAttributesClassName
+	// Only trigger modify volume if the following conditions are met
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) && vacName != nil && *vacName != "" {
```

Comment:
It does if ctrl.resizePVC succeeded without error. Otherwise ``make test`` also complain about here as unreachable code.

Oh the pvcNeedsResize is a good point let me take a look.